libp2p/src/builder.rs

```diff
@@ -373,12 +630,20 @@ mod tests {
     }
 
     #[test]
-    #[cfg(all(feature = "tokio", feature = "tcp", feature = "dns"))]
+    #[cfg(all(
+        feature = "tokio",
+        feature = "tcp",
+        feature = "tls",
+        feature = "noise",
+        feature = "dns"
+    ))]
     fn tcp_dns() {
         let _: libp2p_swarm::Swarm<libp2p_swarm::dummy::Behaviour> = SwarmBuilder::new()
             .with_new_identity()
             .with_tokio()
             .with_tcp()
+            .with_tls()
+            .with_noise()
             .without_relay()
             .no_more_other_transports()
             .with_dns()
```

Comment:
It would probably lead to a bit of duplication but can we add a `with_behaviour` function to the builder after `with_noise` that simply calls these three functions?

Reply:
Added the shortcut. Good idea.

Reply:
Done